### PR TITLE
Update Plone/Python support section.

### PR DESCRIPTION
**Pull request by tisto**

### CODE REVIEW

**Review by mauritsvanrees — APPROVED** [review]
LGTM

File: README.rst [hunk 1]
```diff
@@ -315,16 +315,11 @@ and then running "bin/buildout"
...
+plone.rest 4.x.x supports Plone 5.2 and 6.x on Python 3.8 and newer.
 
-plone.restapi supports Python 2.7 and 3.8 for Plone 5.2 and Python 3.8, 3.9, 3.10, and 3.11 for Plone 6.
-
-Older versions of Python and Plone most likely will continue to work with plone.rest.
-
-Though, we do not test or officially support them.
-
-Check older versions of plone.rest for official support.
+plone.rest 3.x.x supports Plone 5.2 and 6.0 on Python 2.7 (Plone 5.2 only) and on Python 3.7 to 3.11.
```
Comment (stevepiercy) [inline README.rst:320]: This is more clear, except I don't know the Python version support for Plone 5.2 off the top of my head.

```suggestion
plone.rest 3.x.x supports Plone 5.2 on Python 2.7 to 3.?? and Plone 6.0 on Python 3.7 to 3.11.
```

File: README.rst [hunk 2]
```diff
@@ -315,16 +315,11 @@ and then running "bin/buildout"
...
 
+If you need to use Plone 4.3, 5.0, and 5.1 on Python 2.7, checkout out plone.rest 2.x.x and 1.x.x.
```
Comment (stevepiercy) [inline README.rst:322]:
```suggestion
If you need to use Plone 4.3, 5.0, or 5.1 on Python 2.7, check out plone.rest 2.x.x or 1.x.x.
```

**Review by stevepiercy — CHANGES_REQUESTED** [review]

File: README.rst [hunk 3]
```diff
@@ -315,16 +315,11 @@ and then running "bin/buildout"
...
+plone.rest 4.x.x supports Plone 5.2 and 6.x on Python 3.8 and newer.
 
-plone.restapi supports Python 2.7 and 3.8 for Plone 5.2 and Python 3.8, 3.9, 3.10, and 3.11 for Plone 6.
-
-Older versions of Python and Plone most likely will continue to work with plone.rest.
-
-Though, we do not test or officially support them.
-
-Check older versions of plone.rest for official support.
+plone.rest 3.x.x supports Plone 5.2 on Python 2.7 to 3.8 and Plone 6.0 on Python 3.7 to 3.11.
```
Comment (mauritsvanrees) [inline README.rst:320]:
```suggestion
plone.rest 3.x.x supports Plone 5.2 on Python 2.7 and 3.6 to 3.8 and Plone 6.0 on Python 3.8 to 3.11.
```

Comment (tisto) [reply README.rst:320]: @mauritsvanrees man, how could I get this wrong. Good catch! Thanks Maurits!

**Pull request merged**